code/game/objects/structures/vehicles/wizmobile.dm

```diff
@@ -69,6 +69,36 @@
 
 	update_mob()
 
+
+/obj/structure/stool/bed/chair/vehicle/wizmobile/Bump(var/atom/obstacle)
+
+	/*												most likely a bad idea
+	if(istype(obstacle, /obj/structure/window/))
+		obstacle.Destroy(brokenup = 1)
+
+	if(istype(obstacle, /obj/structure/grille/))
+		var/obj/structure/grille/G = obstacle
+		G.health = (0.25*initial(G.health))
+		G.broken = 1
+		G.icon_state = "[initial(G.icon_state)]-b"
+		G.density = 0
+		getFromPool(/obj/item/stack/rods, get_turf(G.loc))
+	*/
+
+	if(istype(obstacle, /obj/structure/table))
+		var/obj/structure/table/T = obstacle
+		T.destroy()
+
+	if(istype(obstacle, /obj/structure/rack))
+		new /obj/item/weapon/rack_parts(obstacle.loc)
```

Comment:
You can do the same as you did for tables for racks.
They also have a destroy() proc that generates their parts.